### DISCUSSION TITLE
Qt 程序依赖库的 Qt 版本疑问

### DISCUSSION THREAD

**Discussion by Ocrosoft** [xmake-io/xmake#1993] [Q&A]
ProA 依赖一个远程 xmake 包 PacA，编译 ProA 的时候指定了 --qt_sdkver
编译 ProA 的时候，拉取并编译 PacA，请问 PacA 的编译会继承指定的 Qt 版本吗？
PS：本地安装了多个 Qt 版本，且通过 --qt_sdkver 指定的版本不是 xmake 默认查找到的版本。

**Comment by waruqi** [comment]
目前没透传 你可以提个 pr 在这里加上

https://github.com/xmake-io/xmake/blob/1658c0de500976efbad38fca52082baec3137fc0/xmake/modules/package/tools/xmake.lua#L85

**Reply by Ocrosoft** [reply]
假设确实使用了同一个 Qt 版本编译，依赖库安装后的作为文件夹名称的 hash 是如何计算的？Qt 的版本是否会纳入计算？
如果不会的话，安装后不同 Qt 版本的项目引用版本和配置都相同的同一个库时，还是可能出现版本问题。

**Reply by waruqi** [reply]
默认不会，目前就 `add_requires` 里面的 `configs = {}` 会参与 buildhash 计算，你可以自己的包里面通过 add_configs 定义几个 config，比如 qtsdkver 等等，然后 xmake.lua 传入 configs = {qtsdkver = get_config("qt_sdkver")} 到 add_requires ，根据 sdk ver 区分下，就行了。。
